src/nominatim_api/results.py

```diff
@@ -225,23 +220,44 @@ def lon(self) -> float:
         """
         return self.centroid[0]
 
+    @property
+    def display_name(self) -> Optional[str]:
+        """ Dynamically compute the display name for the result place
+            and, if available, its address information..
+        """
+        # No longer sets the locale name, want to do that explicitly in format.py
+        # using the formatter object
+        if self.address_rows:  # if this is true we need additional processing
+            label_parts: List[str] = []
+
+            for line in self.address_rows:  # assume locale_name is set by external formatter
+                if not line.local_name:
+                    line.local_name = line.display_name  # fallback to populate
```

Comment:
You don't need that complicated if construct. Just use line.display_name below and it will return you the right thing. (Just don't forget to check that it isn't empty/None and skip over in that case).